uint8 position
int8 speed